modules/react/table/lib/Table.tsx

```diff
@@ -1,29 +1,137 @@
 import React from 'react';
-import {GridProps, mergeStyles} from '@workday/canvas-kit-react/layout';
+import {GridProps, FlexProps, mergeStyles} from '@workday/canvas-kit-react/layout';
 import {createComponent} from '@workday/canvas-kit-react/common';
-import {TableHead} from './TableHead';
-import {TableBody} from './TableBody';
+import {BaseTable, baseTableStencil} from './BaseTable';
 import {TableRow} from './TableRow';
-import {TableHeader} from './TableHeader';
-import {TableCell} from './TableCell';
-import {TableFooter} from './TableFooter';
-import {TableCaption} from './TableCaption';
 import {createStencil, px2rem} from '@workday/canvas-kit-styling';
+import {baseTableBodyStencil} from './BaseTableBody';
+import {baseTableCaptionStencil} from './BaseTableCaption';
+import {baseTableCellStencil} from './BaseTableCell';
+import {baseTableHeadStencil} from './BaseTableHead';
+import {baseTableHeaderStencil} from './BaseTableHeader';
 import {system} from '@workday/canvas-tokens-web';
 
+export interface TableProps extends GridProps {}
+
 const tableStencil = createStencil({
+  extends: baseTableStencil,
+  base: {
+    display: 'grid',
+  },
+});
+
+const tableBodyStencil = createStencil({
+  extends: baseTableBodyStencil,
+  base: {
+    display: 'grid',
+  },
+});
+
+const tableCaptionStencil = createStencil({
+  extends: baseTableCaptionStencil,
+  base: {
+    display: 'flex',
+    borderBottom: `${px2rem(1)} solid ${system.color.border.container}`,
+  },
+});
+
+const tableCellStencil = createStencil({
+  extends: baseTableCellStencil,
+  base: {
+    display: 'grid',
+  },
+});
+
+const tableFooterStencil = createStencil({
   base: {
-    ...system.type.subtext.large,
     display: 'grid',
-    border: `${px2rem(1)} solid ${system.color.border.container}`,
-    borderRadius: system.shape.x2,
-    overflow: 'auto',
-    color: system.color.text.default,
+  },
+});
+
+const tableHeadStencil = createStencil({
+  extends: baseTableHeadStencil,
+  base: {
+    display: 'grid',
+  },
+});
+
+const tableHeaderStencil = createStencil({
+  extends: baseTableHeaderStencil,
+  base: {
+    display: 'grid',
+    alignItems: 'center',
+  },
+});
+
+export const TableBody = createComponent('tbody')({
+  displayName: 'Table.Body',
+  Component: ({children, ...elemProps}: GridProps, ref, Element) => {
+    return (
+      <Element ref={ref} {...mergeStyles(elemProps, tableBodyStencil())}>
+        {children}
+      </Element>
+    );
+  },
+});
+
+export const TableCaption = createComponent('caption')({
+  displayName: 'Table.Caption',
+  Component: ({children, ...elemProps}: FlexProps, ref, Element) => {
+    return (
+      <Element ref={ref} {...mergeStyles(elemProps, tableCaptionStencil())}>
```

Comment:
Question for @NicholasBoll . If `TableCaption` extends `baseTableCaption` should this also render `BaseTableCaption`?

Reply:
You mean in terms of rendering `<Element/>` or `<BaseTableCaption/>`